app/src/main/java/nu/yona/app/YonaApplication.java

```diff
@@ -202,5 +212,54 @@ public static Tracker getTracker()
 		}
 		return mContext.tracker;
 	}
+
+
+	private ServiceConnection getVpnServiceConnection()
+	{
+		return connection = new ServiceConnection()
+		{
+			@Override
+			public void onServiceConnected(ComponentName className,
+										   IBinder service)
+			{
+				// We've bound to LocalService, cast the IBinder and get LocalService instance
+				OpenVPNService.LocalBinder binder = (OpenVPNService.LocalBinder) service;
+				vpnService = binder.getService();
+				stopVPN();
+				getAppContext().unbindService(connection);
+			}
+
+			@Override
+			public void onServiceDisconnected(ComponentName arg0)
+			{
+				vpnService = null;
+				// TODO: Need to show user a alert on failure to stop vpn.
+				nu.yona.app.utils.Logger.loge(YonaApplication.class, "Error in binding the vpn service with " + arg0.getClassName());
+			}
+
+		};
+	}
+
+	private void stopVPN()
```

Comment:
Make this part of the anonymous class and add the VPN service as a parameter, so you don't need a member variable for it.